### ISSUE TITLE
Rust-1.77.0 fails to build with 18.1.2: need LLVM sources available to build `compiler-rt`, but they weren't present; consider enabling `build.submodules = true` or disabling `optimized-compiler-builtins`

### ISSUE THREAD

**Issue by perfect7gentleman** [rust-lang/rust#122946]
```
running: /tmp/portage/dev-lang/rust-1.77.0/work/rustc-1.77.0-src/build/bootstrap/debug/bootstrap build -vvv --config=/tmp/portage/dev-lang/rust-1.77.0/work/rustc-1.77.0-src/config.toml -j9
WARNING: The use of `changelog-seen` is deprecated. Please refer to `change-id` option in `config.example.toml` instead.
WARNING: The `change-id` is missing in the `config.toml`. This means that you will not be able to track the major changes made to the bootstrap configurations.
NOTE: to silence this warning, add `change-id = 102579` at the top of `config.toml`
auto-detected local-rebuild 1.77.0
finding compilers
CC_x86_64-unknown-linux-gnu = "clang-18"
CFLAGS_x86_64-unknown-linux-gnu = ["-ffunction-sections", "-fdata-sections", "-fPIC", "--target=x86_64-unknown-linux-gnu", "-m64"]
CXX_x86_64-unknown-linux-gnu = "clang++-18"
CXXFLAGS_x86_64-unknown-linux-gnu = ["-ffunction-sections", "-fdata-sections", "-fPIC", "--target=x86_64-unknown-linux-gnu", "-march=native", "-mtune=native", "-pipe", "-mllvm", "-polly", "-fforce-emit-vtables", "-femit-dwarf-unwind=no-compact-unwind", "-fsplit-lto-unit", "-fomit-frame-pointer", "-fopenmp=libomp", "-fstrict-enums", "-fstrict-vtable-pointers", "-fuse-ld=lld", "-fno-asynchronous-unwind-tables", "-fno-common", "-fno-plt", "-fno-stack-protector", "-fno-standalone-debug", "-fno-unwind-tables", "-g0", "-w", "-Wno-error=int-conversion", "-Wno-error=implicit-function-declaration", "-Wno-error=implicit-int", "-Wno-error=incompatible-function-pointer-types", "-stdlib=libc++"]
AR_x86_64-unknown-linux-gnu = "/usr/lib/llvm/18/bin/llvm-ar"
running sanity check
learning about cargo
> Assemble { target_compiler: Compiler { stage: 2, host: x86_64-unknown-linux-gnu } }
  > Assemble { target_compiler: Compiler { stage: 1, host: x86_64-unknown-linux-gnu } }
    > Assemble { target_compiler: Compiler { stage: 0, host: x86_64-unknown-linux-gnu } }
    < Assemble { target_compiler: Compiler { stage: 0, host: x86_64-unknown-linux-gnu } }
    > Rustc { target: x86_64-unknown-linux-gnu, compiler: Compiler { stage: 0, host: x86_64-unknown-linux-gnu }, crates: [] }
      > Std { target: x86_64-unknown-linux-gnu, compiler: Compiler { stage: 0, host: x86_64-unknown-linux-gnu }, crates: [], force_recompile: false, extra_rust_args: [], is_for_mir_opt_tests: false }
        > StartupObjects { compiler: Compiler { stage: 0, host: x86_64-unknown-linux-gnu }, target: x86_64-unknown-linux-gnu }
        < StartupObjects { compiler: Compiler { stage: 0, host: x86_64-unknown-linux-gnu }, target: x86_64-unknown-linux-gnu }
        c Assemble { target_compiler: Compiler { stage: 0, host: x86_64-unknown-linux-gnu } }
        > Libdir { compiler: Compiler { stage: 0, host: x86_64-unknown-linux-gnu }, target: x86_64-unknown-linux-gnu }
          > Sysroot { compiler: Compiler { stage: 0, host: x86_64-unknown-linux-gnu }, force_recompile: false }
Removing sysroot /tmp/portage/dev-lang/rust-1.77.0/work/rustc-1.77.0-src/build/x86_64-unknown-linux-gnu/stage0-sysroot to avoid caching bugs
          < Sysroot { compiler: Compiler { stage: 0, host: x86_64-unknown-linux-gnu }, force_recompile: false }
Removing sysroot /tmp/portage/dev-lang/rust-1.77.0/work/rustc-1.77.0-src/build/x86_64-unknown-linux-gnu/stage0-sysroot/lib/rustlib/x86_64-unknown-linux-gnu/lib to avoid caching bugs
          c Sysroot { compiler: Compiler { stage: 0, host: x86_64-unknown-linux-gnu }, force_recompile: false }
        < Libdir { compiler: Compiler { stage: 0, host: x86_64-unknown-linux-gnu }, target: x86_64-unknown-linux-gnu }
        c Libdir { compiler: Compiler { stage: 0, host: x86_64-unknown-linux-gnu }, target: x86_64-unknown-linux-gnu }
        c Libdir { compiler: Compiler { stage: 0, host: x86_64-unknown-linux-gnu }, target: x86_64-unknown-linux-gnu }
        c Sysroot { compiler: Compiler { stage: 0, host: x86_64-unknown-linux-gnu }, force_recompile: false }
thread 'main' panicked at src/core/build_steps/compile.rs:431:13:
need LLVM sources available to build `compiler-rt`, but they weren't present; consider enabling `build.submodules = true` or disabling `optimized-compiler-builtins`
stack backtrace:
   0: rust_begin_unwind
             at /rustc/aedd173a2c086e558c2b66d3743b344f977621a7/library/std/src/panicking.rs:647:5
   1: core::panicking::panic_fmt
             at /rustc/aedd173a2c086e558c2b66d3743b344f977621a7/library/core/src/panicking.rs:72:14
   2: bootstrap::core::build_steps::compile::std_cargo
   3: <bootstrap::core::build_steps::compile::Std as bootstrap::core::builder::Step>::run
   4: bootstrap::core::builder::Builder::ensure
   5: <bootstrap::core::build_steps::compile::Rustc as bootstrap::core::builder::Step>::run
   6: bootstrap::core::builder::Builder::ensure
   7: <bootstrap::core::build_steps::compile::Assemble as bootstrap::core::builder::Step>::run
   8: bootstrap::core::builder::Builder::ensure
   9: <bootstrap::core::build_steps::compile::Assemble as bootstrap::core::builder::Step>::run
  10: bootstrap::core::builder::Builder::ensure
  11: <bootstrap::core::build_steps::compile::Std as bootstrap::core::builder::Step>::make_run
  12: bootstrap::core::builder::StepDescription::maybe_run
  13: bootstrap::core::builder::Builder::run_step_descriptions
  14: bootstrap::Build::build
  15: bootstrap::main
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
Traceback (most recent call last):
  File "/tmp/portage/dev-lang/rust-1.77.0/work/rustc-1.77.0-src/./x.py", line 50, in <module>
    bootstrap.main()
  File "/tmp/portage/dev-lang/rust-1.77.0/work/rustc-1.77.0-src/src/bootstrap/bootstrap.py", line 1162, in main
    bootstrap(args)
  File "/tmp/portage/dev-lang/rust-1.77.0/work/rustc-1.77.0-src/src/bootstrap/bootstrap.py", line 1138, in bootstrap
    run(args, env=env, verbose=build.verbose, is_bootstrap=True)
  File "/tmp/portage/dev-lang/rust-1.77.0/work/rustc-1.77.0-src/src/bootstrap/bootstrap.py", line 187, in run
    raise RuntimeError(err)
RuntimeError: failed to run: /tmp/portage/dev-lang/rust-1.77.0/work/rustc-1.77.0-src/build/bootstrap/debug/bootstrap build -vvv --config=/tmp/portage/dev-lang/rust-1.77.0/work/rustc-1.77.0-src/config.toml -j9
```
```
[build]
build-stage = 2
test-stage = 2
build = "x86_64-unknown-linux-gnu"
host = ["x86_64-unknown-linux-gnu"]
target = ["x86_64-unknown-linux-gnu"]
cargo = "/opt/rust-bin-1.77.0/bin/cargo"
rustc = "/opt/rust-bin-1.77.0/bin/rustc"
rustfmt = "/opt/rust-bin-1.77.0/bin/rustfmt"
docs = false
compiler-docs = false
#
submodules = true
optimized-compiler-builtins = true
#
python = "python3.12"
locked-deps = false
vendor = true
extended = true
tools = ["cargo","rustdoc"]
verbose = 2
sanitizers = false
profiler = false
cargo-native-static = false
local-rebuild = false
```
```
[llvm.build-config]
CMAKE_VERBOSE_MAKEFILE = "ON"
CMAKE_C_FLAGS_RELEASE = "-march=native -mtune=native -O3 -pipe -mllvm -polly -fforce-emit-vtables -femit-dwarf-unwind=no-compact-unwind -fsplit-lto-unit -fomit-frame-pointer -fopenmp=libomp -fstrict-enums -fstrict-vtable-pointers -fuse-ld=lld -fno-asynchronous-unwind-tables -fno-common -fno-plt -fno-stack-protector -fno-standalone-debug -fno-unwind-tables -g0 -w -Wno-error=int-conversion -Wno-error=implicit-function-declaration -Wno-error=implicit-int -Wno-error=incompatible-function-pointer-types"
CMAKE_CXX_FLAGS_RELEASE = "-march=native -mtune=native -O3 -pipe -mllvm -polly -fforce-emit-vtables -femit-dwarf-unwind=no-compact-unwind -fsplit-lto-unit -fomit-frame-pointer -fopenmp=libomp -fstrict-enums -fstrict-vtable-pointers -fuse-ld=lld -fno-asynchronous-unwind-tables -fno-common -fno-plt -fno-stack-protector -fno-standalone-debug -fno-unwind-tables -g0 -w -Wno-error=int-conversion -Wno-error=implicit-function-declaration -Wno-error=implicit-int -Wno-error=incompatible-function-pointer-types -stdlib=libc++"
CMAKE_EXE_LINKER_FLAGS_RELEASE = "-Wl,-O2 -Wl,--as-needed -Wl,--gc-sections -Wl,--icf=all -Wl,--lto-O3 -Wl,--thinlto-jobs=9 -Wl,-z,norelro -lomp -rtlib=compiler-rt -unwindlib=libunwind -s -march=native -mtune=native -O3 -pipe -mllvm -polly -fforce-emit-vtables -femit-dwarf-unwind=no-compact-unwind -fsplit-lto-unit -fomit-frame-pointer -fopenmp=libomp -fstrict-enums -fstrict-vtable-pointers -fuse-ld=lld -fno-asynchronous-unwind-tables -fno-common -fno-plt -fno-stack-protector -fno-standalone-debug -fno-unwind-tables -g0 -w -Wno-error=int-conversion -Wno-error=implicit-function-declaration -Wno-error=implicit-int -Wno-error=incompatible-function-pointer-types -stdlib=libc++"
CMAKE_MODULE_LINKER_FLAGS_RELEASE = "-Wl,-O2 -Wl,--as-needed -Wl,--gc-sections -Wl,--icf=all -Wl,--lto-O3 -Wl,--thinlto-jobs=9 -Wl,-z,norelro -lomp -rtlib=compiler-rt -unwindlib=libunwind -s -march=native -mtune=native -O3 -pipe -mllvm -polly -fforce-emit-vtables -femit-dwarf-unwind=no-compact-unwind -fsplit-lto-unit -fomit-frame-pointer -fopenmp=libomp -fstrict-enums -fstrict-vtable-pointers -fuse-ld=lld -fno-asynchronous-unwind-tables -fno-common -fno-plt -fno-stack-protector -fno-standalone-debug -fno-unwind-tables -g0 -w -Wno-error=int-conversion -Wno-error=implicit-function-declaration -Wno-error=implicit-int -Wno-error=incompatible-function-pointer-types -stdlib=libc++"
CMAKE_SHARED_LINKER_FLAGS_RELEASE = "-Wl,-O2 -Wl,--as-needed -Wl,--gc-sections -Wl,--icf=all -Wl,--lto-O3 -Wl,--thinlto-jobs=9 -Wl,-z,norelro -lomp -rtlib=compiler-rt -unwindlib=libunwind -s -march=native -mtune=native -O3 -pipe -mllvm -polly -fforce-emit-vtables -femit-dwarf-unwind=no-compact-unwind -fsplit-lto-unit -fomit-frame-pointer -fopenmp=libomp -fstrict-enums -fstrict-vtable-pointers -fuse-ld=lld -fno-asynchronous-unwind-tables -fno-common -fno-plt -fno-stack-protector -fno-standalone-debug -fno-unwind-tables -g0 -w -Wno-error=int-conversion -Wno-error=implicit-function-declaration -Wno-error=implicit-int -Wno-error=incompatible-function-pointer-types -stdlib=libc++"

```
```
~ $ equery f compiler-rt
 * Searching for compiler-rt ...
 * Contents of sys-libs/compiler-rt-18.1.2:
/usr
/usr/lib
/usr/lib/clang
/usr/lib/clang/18
/usr/lib/clang/18/lib
/usr/lib/clang/18/lib/linux
/usr/lib/clang/18/lib/linux/clang_rt.crtbegin-x86_64.o
/usr/lib/clang/18/lib/linux/clang_rt.crtend-x86_64.o
/usr/lib/clang/18/lib/linux/libclang_rt.builtins-x86_64.a
/usr/share
/usr/share/doc
/usr/share/doc/compiler-rt-18.1.2
/usr/share/doc/compiler-rt-18.1.2/README.txt.bz2
```

**Comment by onur-ozkan** [comment]
The error message gives enough information already. `optimized-compiler-builtins` option needs to build "compiler-rt" which is in the "llvm-project" submodule.
